server/historian/packages/historian-base/src/services/restGitService.ts

```diff
@@ -110,10 +116,16 @@ export class RestGitService {
     }
 
     public async createRef(params: ICreateRefParamsExternal): Promise<git.IRef> {
+        if (!this.writeToExternalStorage) {
+            params.config.enabled = false;
```

Comment:
Don't we have to do this for the getRef flow too?

Reply:
I don't like mutating a param. But I copying and creating a new param also looks ugly. To reduce the ugliness, we should write comment on why we are doing this.